geoportal_1/src/main/java/org/opengeoportal/ogc/wcs/wcs1_0_0/WcsGetCoverage1_0_0.java

```diff
@@ -46,23 +49,67 @@ public static String createWcsGetCoverageRequest(String layerName, CoverageOffer
+		    //The envelope is then transformed to the native (targetCRS) bounds below.		    
 		    ReferencedEnvelope envelope = new ReferencedEnvelope(bounds.getMinX(), bounds.getMaxX(), bounds.getMinY(), bounds.getMaxY(), sourceCRS);
 		    logger.info("requested Bounds: " + OgpUtils.referencedEnvelopeToString(envelope));
+		    
 		    // Transform using 10 sample points around the envelope
-
 		    CoordinateReferenceSystem targetCRS = factory.createCoordinateReferenceSystem(coverageOffering.getSupportedCRSs().get(0));
 
 		    ReferencedEnvelope result = envelope.transform(targetCRS, true, 10);
 
-		 String getCoverageRequest = "service=WCS&version=" + VERSION + 
+		ReferencedEnvelope nativeRefEnv = new ReferencedEnvelope(nativeBbox.getMinX(), nativeBbox.getMaxX(), nativeBbox.getMinY(), nativeBbox.getMaxY(), targetCRS);
+		
+		Double resX = coverageOffering.getRectifiedGrid().getResx();
+		Double resY = coverageOffering.getRectifiedGrid().getResy();
+
+		Double xMin, yMin, xMax, yMax;
+
+		if ( nativeRefEnv.getMinimum(0) > result.getMinimum(0)){
+			xMin = nativeRefEnv.getMinimum(0);
+		} else {
+			Double x_a = (Math.abs(result.getMinimum(0) - nativeRefEnv.getMinimum(0)))/resX;
+			xMin = nativeRefEnv.getMinimum(0) + (x_a.intValue() * resX);    }
+
+		if ( nativeRefEnv.getMinimum(1) > result.getMinimum(1)){
+			yMin = nativeRefEnv.getMinimum(1);
+		} else {
+			Double y_a = (Math.abs(result.getMinimum(1) - nativeRefEnv.getMinimum(1)))/resY;
+			yMin = nativeRefEnv.getMinimum(1) + (y_a.intValue() * resY);    }
+
+
+		if ( nativeRefEnv.getMaximum(0) < result.getMaximum(0)){
+			xMax = nativeRefEnv.getMaximum(0);
+		} else {
+			Double x_b = (Math.abs(result.getMaximum(0) - nativeRefEnv.getMaximum(0)))/resX;
+			xMax = nativeRefEnv.getMaximum(0) - (x_b.intValue() * resX);    }
+
+
+		if ( nativeRefEnv.getMaximum(1) < result.getMaximum(1)){
+			yMax = nativeRefEnv.getMaximum(1);
+		} else {
+			Double y_b = (Math.abs(result.getMaximum(1) - nativeRefEnv.getMaximum(1)))/resY;
+			yMax =  nativeRefEnv.getMaximum(1) - (y_b.intValue() * resY);   }
+
+		String bbox = "&bbox=" + Double.toString(xMin) + "," + Double.toString(yMin) + "," + Double.toString(xMax) + "," + Double.toString(yMax);
+		Double dwidth = (Math.abs(xMax - xMin))/resX;
+		Double dheight = (Math.abs(yMax - yMin))/resY;
+		int width = dwidth.intValue();
+		int height = dheight.intValue();
+		
+		String widthHeight = "&width=" + width + "&height=" + height;
+		
+		    String getCoverageRequest = "service=WCS&version=" + VERSION + 
 		 		"&request=GetCoverage&coverage=" + layerName +
-				"&bbox=" + OgpUtils.referencedEnvelopeToString(result) +
+				bbox +				
+				//"&bbox=" + OgpUtils.referencedEnvelopeToString(result) +
+				// "&bbox=" + OgpUtils.referencedEnvelopeToString(result) +
 				"&crs=" + coverageOffering.getSupportedCRSs().get(0) + 
-				"&format=" + outputFormat;
-		 
-		getCoverageRequest += generateSize(coverageOffering.getRectifiedGrid());
```

Comment:
No longer need generateSize. The getCoverageRequest string is composed near the end of the file.